galaxychop/galaxychop.py

```diff
@@ -148,13 +143,21 @@ class Galaxy:
     pot_dark = np.loadtxt(path_potencial+'dark_ID_'+str(ID[j])+'.dat')
     pot_gas = np.loadtxt(path_potencial+'gas_ID_'+str(ID[j])+'.dat')
 
-    # Calculamos la energia.
-    E_tot_star = k_star - pot_star[:, 1]
-    E_tot_dark = k_dark - pot_dark[:, 1]
-    E_tot_gas = k_gas - pot_gas[:, 1]
+    def energy(self, pot_star, pot_dark, pot_gas):
+
+        '''Calculamos la energia cinetica y potencial y total'''
+
+        k_s = 0.5*(self.vx_s**2 + self.vy_s**2 + self.vz_s**2)
+        k_dm = 0.5*(self.vx_dm**2 + self.vy_dm**2 + self.vz_dm**2)
+        k_g = 0.5*(self.vx_g**2 + self.vy_g**2 + self.vz_g**2)
+
+        E_tot_star = k_s - self.pot_star[:, 1]
+        E_tot_dark = k_dm - self.pot_dark[:, 1]
+        E_tot_gas = k_g - self.pot_gas[:, 1]
+
+        E_tot = np.concatenate((E_tot_gas, E_tot_dark, E_tot_star))
 
-    E_tot = np.concatenate((E_tot_gas, E_tot_dark, E_tot_star))
-    
+        return E_tot, E_tot_star, E_tot_dark, E_tot_gas
```

Comment:
Is it ok to leave this reurning twice the same outputs?

I would say that probably we would like to store `E_tot` in the attributes, but I am not aware of design.
If attributes are the choice here, probably storing them as `self.E_comp` is sensible.

Reply:
I think the method should return only `E_tot_gas`, `E_tot_dark`, `E_tot_star`. Because `E_tot` of all particles is only used for the calculation of `J_circ`, and ultimately `E_tot_gas` would serve in case you want to do the dynamic decomposition of the gas.

Reply:
Ok, this method will return E_tot_gas, E_tot_dark and E_tot_star for now.